sway-parse/src/token.rs

```diff
@@ -159,7 +159,7 @@ pub fn lex_commented(
                         .count()
                         > 0;
 
-                    let comment_kind = if has_newline {
+                    let comment_kind = if has_newline || (search_end == 0 && index == 0) {
```

Comment:
Hmm not sure I got why we needed this. Is it for detecting comments at the beginning of the file as `newlined` or somehting else? If it is not something obvious that I am missing rn and you would like, maybe could use a single comment 👍

Reply:
>  Is it for detecting comments at the beginning of the file

Yes! Will add a comment to this